homeassistant/components/manual_mqtt/alarm_control_panel.py

```diff
@@ -406,20 +467,39 @@ def _async_validate_code(self, code, state):
     @property
     def extra_state_attributes(self) -> dict[str, Any]:
         """Return the state attributes."""
-        if self.state != STATE_ALARM_PENDING:
-            return {}
-        return {
-            ATTR_PRE_PENDING_STATE: self._previous_state,
-            ATTR_POST_PENDING_STATE: self._state,
-        }
+        if self.state in (STATE_ALARM_PENDING, STATE_ALARM_ARMING):
+            return {
+                ATTR_PREVIOUS_STATE: self._previous_state,
```

Comment:
Is it really necessary to change these attributes and break.

Reply:
Unfortunately yes, it is. The real motivation for me to do this PR is that I would like to have alarm control panel groups (reviving my old work from #11323), and the group needs some consistency between the devices that they include. While the group code I have written can deal with a control panel that doesn't have previous/next state attributes, the behavior is inferior.

This also justifies moving them to const, because they become standardized for all alarm control panels that want to provide full functionality for groups.

I have created a single large pull request because, more than new functionality I saw this work as fixing the mistakes that led "manual" and "manual MQTT" to become unsynchronized (#10697, #17270, #32950 and #55340). It _is_ really easier to review the combined changes than any split PR, but of course I can split it if you prefer.

To be honest, I did wonder if the "manual MQTT" integration should exist at all. It seems just as easy to do it using a "normal" manual control panel plus a template alarm control panel. But it seems to have a couple hundred users according to the telemetry, so I didn't propose the deprecation route.